### PR TITLE
Car class details and first mocking version

### PR DESCRIPTION
**Pull request by elft3r**
Hey @kart7990,
I added a new component with single- and multi-class details. This helped me further learn the data structure even further.

While doing this, I also added a first version of mocking the API. It's not fancy for now and just reads the JSON files in the `mock-data` folder. However, I found this to be the easiest implementation to get going and iterate faster. Happy to hear your thoughts on this and how to improve it. You can use this by running `npm run dev-mock`.

One question that I have is regarding the car details in the standings. For the car class, the API returns the `classId`, `className` and `classColor`. However, for the car, it only returns the `carName`, and not the `carId` or `carColor`. Where do you get the information for the class from? Is this from the iRacing API or are you setting these in the backend? How can we best set this for the cars as well?

Best,
Jochen

### CODE REVIEW

**Comment by elft3r** [comment]
FYI: Both of the class detail components are not yet very efficient. I will fix that in the next iteration of the component. I will then move the calculation to the state update mechanism so that it only gets updated when a new fastest lap time is set.

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "scripts": {
     "dev": "next dev -p 3000",
     "dev-local": "set \"TARGET_ENV=development-localserver\" && next dev -p 3000",
+    "dev-mock": "MOCKING=true next dev -p 3000",
```
Comment (kart7990) [inline package.json]: I wasn't able to get this to work. I had to modify it to `"dev-mock": "set \"MOCKING=true\" && next dev -p 3000",`

Am I missing something?

Fwiw, I'm on a windows machine.

File: next.config.js [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /** @type {import('next').NextConfig} */
 
+const { env } = require("process");
```
Comment (kart7990) [inline next.config.js]: Is this needed?

File: src/components/car-class-details/single-class.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+interface SplitDetails {
+  iRatingTotal: number;
+  count: number;
+}
+
+export const SingleClassDetails = () => {
+  const session = useSelector(selectCurrentSession);
+  const standings = useSelector<LiveTiming[]>(selectLiveTiming);
```
Comment (kart7990) [inline src/components/car-class-details/single-class.tsx:28]: Yeah this is going to be lots of calculations every half second (or whatever update interval we decide on for standings). I have no prob merging as is and optimize later, though.

File: src/app/pitwall/dashboard/[id]/page.tsx [hunk 4]
```diff
@@ -1,59 +1,29 @@
...
+      return (
+        <PitwallSessionMock pitwallSessionId={pitboxSessionId}>
+          <Dashboard />
+        </PitwallSessionMock>
+      );
+    } else {
+      return (
+        <PitwallSession pitwallSessionId={pitboxSessionId}>
+          <Dashboard />
+        </PitwallSession>
```
Comment (kart7990) [inline src/app/pitwall/dashboard/[id]/page.tsx:23]: Nice, love how clean this is. Even if the mock data is static, still a step forward!

File: package.json (hunk 1 above)
Comment (elft3r) [reply package.json]: I changed it now by adding `cross-env`. That's a problem between different OS platforms. But with this little tool, it should work on all OS platforms.

File: next.config.js (hunk 2 above)
Comment (elft3r) [reply next.config.js]: No, I removed it. There is still a bug in the eslint and prettier config. It does not yet properly remove not needed imports. Or only in some cases. It's something I still have to figure out how to solve this.

File: src/app/pitwall/dashboard/[id]/page.tsx (hunk 4 above)
Comment (elft3r) [reply src/app/pitwall/dashboard/[id]/page.tsx:23]: Yes, I'm not yet fully happy with it. As it also means that we have the mock in the final app. I will try to dig a bit deeper and find a nice way to inject the mock on startup.

Comment (kart7990) [reply src/app/pitwall/dashboard/[id]/page.tsx:23]: True, not ideal to have mock data logic in release builds. All good for now, though!

File: package.json (hunk 1 above)
Comment (kart7990) [reply package.json]: Works like a champ 👍

**Review by kart7990 — APPROVED** [review]

**Pull request merged**